Sources/MongoSwift/BSON/BsonValue.swift

```diff
@@ -185,27 +165,75 @@ public struct Binary: BsonValue, Equatable, Codable {
     public let data: Data
 
     /// The binary subtype for this data.
-    public let subtype: BsonSubtype
+    public let subtype: Subtype
+
+    /// Subtypes for BSON Binary values.
+    public enum Subtype: RawRepresentable, Codable {
+        /// Generic binary subtype
+        case generic,
+        /// A function
+        function,
+        /// Binary (old)
+        binaryDeprecated,
+        /// UUID (old)
+        uuidDeprecated,
+        /// UUID (RFC 4122)
+        uuid,
+        /// MD5
+        md5,
+        /// User defined
+        userDefined(UInt8)
+
+        public init?(rawValue: UInt8) {
+            switch rawValue {
+            case 0: self = .generic
+            case 1: self = .function
+            case 2: self = .binaryDeprecated
+            case 3: self = .uuidDeprecated
+            case 4: self = .uuid
+            case 5: self = .md5
+            case 0x80...0xFF:
+                self = .userDefined(rawValue)
+            default:
+                return nil
```

Comment:
Per my earlier comment, I think we still want to accept any unsigned 8-bit value and only consider 0x80 to be `userDefined` (as the constant is defined in other drivers).